packages/compiler/core/logger.ts

```diff
@@ -46,34 +49,86 @@ function processLog(log: LogInfo): ProcessedLog {
   };
 }
 
-export function createConsoleSink(): LogSink {
+export interface FormatLogOptions {
+  pretty?: boolean;
+}
+
+export interface LogSinkOptions extends FormatLogOptions {}
+
+export function createConsoleSink(options: LogSinkOptions = {}): LogSink {
   function log(data: ProcessedLog) {
     // eslint-disable-next-line no-console
-    console.log(formatLog(data));
+    console.log(formatLog(data, options));
   }
 
   return {
     log,
   };
 }
 
-export function formatLog(log: ProcessedLog): string {
-  const code = log.code ? ` ${log.code}` : "";
-  const level = log.level;
+export function formatLog(log: ProcessedLog, options: FormatLogOptions): string {
+  const code = color(options, log.code ? ` ${log.code}` : "", pc.gray);
+  const level = formatLevel(options, log.level);
   const content = `${level}${code}: ${log.message}`;
   const location = log.sourceLocation;
   if (location?.file) {
-    const formattedLocation = formatSourceLocation(location);
-    return `${formattedLocation} - ${content}`;
+    const formattedLocation = formatSourceLocation(options, location);
+    const sourcePreview = formatSourcePreview(options, location);
+    return `${formattedLocation} - ${content}${sourcePreview}`;
   } else {
     return content;
   }
 }
 
-export function formatSourceLocation(location: SourceLocation) {
+function color(options: FormatLogOptions, text: string, color: Formatter) {
+  return options.pretty ? color(text) : text;
+}
+
+function formatLevel(options: FormatLogOptions, level: LogLevel) {
+  switch (level) {
+    case "error":
+      return color(options, "error", pc.red);
+    case "warning":
+      return color(options, "warning", pc.yellow);
+    default:
+      return level;
+  }
+}
+
+function formatSourceLocation(options: FormatLogOptions, location: SourceLocation) {
+  const postition = getLineAndColumn(location);
+  const path = color(options, location.file.path, pc.cyan);
+
+  const line = color(options, postition.line.toString(), pc.yellow);
+  const column = color(options, postition.column.toString(), pc.yellow);
+  return `${path}:${line}:${column}`;
+}
+
+/**
+ * Create a preview of where the log location is.
+ *
+ * ----------------------------------------------
+ *   4 |
+ *   5 | @route("/alpha/{id}")
+ * > 6 | op doAlpha(@path id: string): abc;
+ *     |                               ^
```

Comment:
Are we able to use the end position so that we get `^^^` or `~~~` under abc rather than just pointing to the start? I think other compilers do that?

Reply:
believe that library does allow that, let me try

Reply:
<img width="911" alt="image" src="https://user-images.githubusercontent.com/1031227/163018742-380b8e2f-8cf0-4d45-8a3f-87349ad15086.png">
works great

Reply:
Yay!